File: data/multilang_comparison/program_go/src/func8.c
#include "program_go.h" 
array_t* func8(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t* array202;
      if (pCounter > 0) {
         array202 = vars->data[--pCounter];
         array202->refC++;
         DEBUG_COPY(array202->id);
      } else {
         array202 = (array_t*)malloc(sizeof(array_t));
         array202->size = 796;
         array202->refC = 1;
         array202->id = 202;
         array202->data = (unsigned int*)malloc(array202->size*sizeof(unsigned int));
         memset(array202->data, 0, array202->size*sizeof(unsigned int));
         DEBUG_NEW(array202->id);
      }
      unsigned int loop59 = 0;
      unsigned int loopLimit59 = (rand()%loopsFactor)/2 + 1;
      for(; loop59 < loopLimit59; loop59++) {
         for (int i = 0; i < array202->size; i++) {
            array202->data[i]--;
         }
         for (int i = 0; i < array202->size; i++) {
            if (array202->data[i] == 80) { 
               return array202;
            }
         }
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array202;
      array_t* array203 = func18(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array203->id);
      free(params0.data);
      array203->refC--;
      if(array203->refC == 0) {
         free(array203->data);
         free(array203);
         DEBUG_FREE(array203->id);
      }
      array202->refC--;
      if(array202->refC == 0) {
         free(array202->data);
         free(array202);
         DEBUG_FREE(array202->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array204 = func14(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array204->id);
      free(params0.data);
      array204->refC--;
      if(array204->refC == 0) {
         free(array204->data);
         free(array204);
         DEBUG_FREE(array204->id);
      }
   }
   array_t* array205;
   if (pCounter > 0) {
      array205 = vars->data[--pCounter];
      array205->refC++;
      DEBUG_COPY(array205->id);
   } else {
      array205 = (array_t*)malloc(sizeof(array_t));
      array205->size = 812;
      array205->refC = 1;
      array205->id = 205;
      array205->data = (unsigned int*)malloc(array205->size*sizeof(unsigned int));
      memset(array205->data, 0, array205->size*sizeof(unsigned int));
      DEBUG_NEW(array205->id);
   }
   return array205;
}

File: data/multilang_comparison/program_go/src/func8.go
package main
import "math/rand"
var _ = rand.Intn
func Func8(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var array170 *Array
      if pCounter > 0 {
          array170 = vars.Data[pCounter-1]
          pCounter--
          array170.RefC++
      } else {
          array170 = &Array{
              Size: 627,
              RefC: 1,
              Id: 170,
              Data: make([]uint32, 627),
          }
      }
      var loop50 int = 0;
      var loopLimit50 int = (50)/2 + 1;
      for ; loop50 < loopLimit50; loop50++ {
         for i := 0; i < array170.Size; i++ {
             array170.Data[i]--
         }
         for i := 0; i < array170.Size; i++ {
             if array170.Data[i] == 14 {
                 return array170
             }
         }
      }
      var params0 ArrayParam
      params0.Size = 1
      params0.Data = make([]*Array, params0.Size)
      params0.Data[0] = array170
      var array171 *Array = Func18(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array171.RefC--
      if array171.RefC == 0 {
          array171.Data = nil
          array171 = nil
      }
      array170.RefC--
      if array170.RefC == 0 {
          array170.Data = nil
          array170 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array172 *Array = Func14(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array172.RefC--
      if array172.RefC == 0 {
          array172.Data = nil
          array172 = nil
      }
   }
   var array173 *Array
   if pCounter > 0 {
       array173 = vars.Data[pCounter-1]
       pCounter--
       array173.RefC++
   } else {
       array173 = &Array{
           Size: 479,
           RefC: 1,
           Id: 173,
           Data: make([]uint32, 479),
       }
   }
   return array173;
}

